src/components/common/Layout/components/Account/components/AccountModal/hooks/useDisconnect.ts

```diff
@@ -1,37 +1,92 @@
 import { useWeb3React } from "@web3-react/core"
 import useGuild from "components/[guild]/hooks/useGuild"
-import useUser from "components/[guild]/hooks/useUser"
+import useUser, { useUserPublic } from "components/[guild]/hooks/useUser"
 import useShowErrorToast from "hooks/useShowErrorToast"
 import { SignedValdation, useSubmitWithSign } from "hooks/useSubmit"
 import { mutateOptionalAuthSWRKey } from "hooks/useSWRWithOptionalAuth"
 import useToast from "hooks/useToast"
-import { PlatformName } from "types"
+import { PlatformType } from "types"
 import fetcher from "utils/fetcher"
 
-type Data =
-  | {
-      platformName: PlatformName
-    }
-  | {
-      address: string
-    }
-
 const useDisconnect = (onSuccess?: () => void) => {
   const showErrorToast = useShowErrorToast()
-  const { mutate: mutateUser } = useUser()
+  const { account } = useWeb3React()
+  const { mutate: mutateUser, id: userId } = useUser()
+  const { mutate: mutatePublicUser } = useUserPublic()
+  const { id } = useGuild()
+  const toast = useToast()
+
+  const submit = async (signedValidation: SignedValdation) => {
+    const { platformName } = JSON.parse(signedValidation.signedPayload)
+
+    return fetcher(
+      `/v2/users/${userId}/platform-users/${PlatformType[platformName]}`,
+      {
+        method: "DELETE",
+        ...signedValidation,
+      }
+    )
+  }
+
+  return useSubmitWithSign<{ platformId: number }>(submit, {
+    onSuccess: ({ platformId }) => {
+      mutateUser(
+        (prev) => ({
```

Comment:
I think it would have been a great idea! I've removed the public one in https://github.com/agoraxyz/guild.xyz/pull/838/commits/bcdc51581c01f0f7c9d29d6f86d5126080bc6db5 as we discussed on Discord!